### PR TITLE
Move healthcheck into Dockerfile

### PR DESCRIPTION
**Pull request by voruti**
... thus the health check is also available when using a simple `docker run ...`.

Or does something speak against it @wolveix @dieser-niko @ashfire908 ?

### CODE REVIEW

**Comment by ashfire908** [comment]
My concern is (and has been) if this will cause any deployments to automatically restart if they don't successfully start up in 2 minutes (ie still downloading from steam due to slow internet). I tested docker-compose, and I assume docker run is the exact same. I looked into Kubernetes, as there is some k8s resources under `cluster/`. Kubernetes [explicitly ignores](https://github.com/kubernetes/kubernetes/pull/50796) the `HEALTHCHECK` declaration in dockerfiles in favor of `liveness` and `readiness` probes, which is good because Kubernetes would 100% run into the issue I was worrying about if `HEALTHCHECK` mapped to `liveness`. So this looks good to me. Long term, it might make sense to use a combination of [liveness, readiness, and startup probes](https://kubernetes.io/docs/concepts/configuration/liveness-readiness-startup-probes/) to cover the expected lifecycle of a Satisfactory pod. But that's an issue that can be tackled in the future. I might make a stab at this at some point, although I currently just run Satisfactory on a stack of Docker/Compose/Portainer, so I don't have a personal use myself (I use Kubernetes heavily in my day job, though).

tl;dr I think this is good and won't cause any restart issues within the environments of Docker, Docker Compose, or the provided Kubernetes resources.

**Comment by dieser-niko** [comment]
I kind of like the idea of having the health check built into the Dockerfile, I just didn't know it existed.

**Comment by schildbach** [comment]
I generally do this with my own Docker projects and think it is a good idea.

One possible pitfall is that if people run this behind a reverse proxy like traefik, those proxies usually only forward traffic until the container is "healthy", which depending on the healthcheck parameters may take a minute or two.

**Comment by wolveix** [comment]
Thanks :)
